src/Text.mo

```diff
@@ -152,10 +227,16 @@ module {
   /// A pattern `p` describes a sequence of characters. A pattern has one of the following forms:
   ///
   /// * `#char c` matches the single character sequence, `c`.
-  /// * `#predicate p` matches any single character sequence `c` satisfying predicate `p(c)`.
   /// * `#text t` matches multi-character text sequence `t`.
+  /// * `#predicate p` matches any single character sequence `c` satisfying predicate `p(c)`.
   ///
   /// A _match_ for `p` is any sequence of characters matching the pattern `p`.
+  ///
+  /// ```motoko include=import
+  /// let charPattern = #char 'A';
+  /// let textPattern = #text "phrase";
+  /// let predicatePattern : Text.Pattern = #predicate (func(c) { c == 'A' or c == 'B' }); // matches "ABABAB"
```

Comment:
```suggestion
  /// let predicatePattern : Text.Pattern = #predicate (func(c) { c == 'A' or c == 'B' }); // matches "A" or "B"
```
I think that is more accurate, but maybe I'm misremembering.

Reply:
Yep, this seems like a better general example.